lib/emergency_banner/display.rb

```diff
@@ -0,0 +1,34 @@
+module EmergencyBanner
+  class Display
+    def enabled?
+      content && campaign_class && heading
+    end
+
+    def campaign_class
+      content[:campaign_class] if content[:campaign_class].present?
+    end
+
+    def heading
+      content[:heading] if content[:heading].present?
+    end
+
+    def short_description
+      content[:short_description] if content[:short_description].present?
+    end
+
+    def link
+      content[:link] if content[:link].present?
+    end
+
+  private
+
+    def content
+      @data ||= connect.hgetall("emergency_banner")
```

Comment:
Yeah so if the key doesn't exist and it returns nil, you need to memoize that `nil` value so that it doesn't try and request it four times per page view. If `@content` ends up being `nil` due to `hgetall` returning `nil`, the `||=` will mean it tried to fetch it again on the next call. Using `if defined?` means that it checks if the `@content` variable has been defined at all, regardless of whether it's `nil` or not.